src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1473,7 +1473,9 @@ private void RemoveNodeFromDatabase(ClusterOperationContext context, BlittableJs
                     if (rawRecord == null)
                         throw new DatabaseDoesNotExistException($"The database {databaseName} does not exists");
 
-                    if (rawRecord.Topology == null)
+                    DatabaseTopology topology = isSharded ? rawRecord.Sharding.Shards[shardNumber] : rawRecord.Topology;
```

Comment:
be careful not to delete the entire database record if we remove a complete shard